### PR TITLE
Remove dead code [No Ticket]

### PR DESCRIPTION
**Pull request by mfraezz**
## Purpose
Remove dead code

## Changes
* Delete `website/addons`
* Remove references to deleted code
* Refactor addon `AppConfig`s to not touch models until they're loaded
* Update template routing
* Deprecate misc. mongo-related code that was affected by `website/addon` removal

## Side effects
None expected. Tested addon widgets locally to ensure routing still worked.

## Ticket
None known

### CODE REVIEW

File: website/citations/models.py [hunk 1]
```diff
@@ -1,131 +0,0 @@
...
-        self.save()
-
-    def after_delete(self, node=None, user=None):
-        self.deauthorize(Auth(user=user), add_log=True)
-
-    def on_delete(self):
-        self.deauthorize(add_log=False)
-
-
-class CitationStyle(StoredObject):
```
Comment (cwisecarver) [inline website/citations/models.py]: This doesn't need to be here for the monkey patching?

File: website/citations/views.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 from framework.auth.decorators import must_be_logged_in
 
-from website.models import CitationStyle
+from osf.models.citation import CitationStyle
```
Comment (cwisecarver) [inline website/citations/views.py]: I guess it doesn't if you fixed the imports.

File: website/citations/models.py (hunk 1 above)
Comment (mfraezz) [inline website/citations/models.py]: Old models have to be phased out eventually. This one was being troublesome, so it got the axe. 
Also,
> cwisecarver
> I guess it doesn't if you fixed the imports.

File: website/app.py [hunk 3]
```diff
@@ -137,8 +124,6 @@ def init_app(settings_module='website.settings', set_backends=True, routes=True,
     app.config['SESSION_COOKIE_SECURE'] = settings.SESSION_COOKIE_SECURE
     app.config['SESSION_COOKIE_HTTPONLY'] = settings.SESSION_COOKIE_HTTPONLY
 
-    if set_backends:
-        do_set_backends(settings)
```
Comment (sloria) [inline website/app.py]: Might be good to raise a DeprecationWarning if `set_backends` is set, but it's not completely necessary.

Comment (mfraezz) [inline website/app.py]: It's still needed below for the `if set_backends and fixtures` block -- lots of tests break if that logic changes

**Pull request merged**